docs/API.md

```diff
@@ -405,1210 +432,64 @@ the epoll container.
+and default transmission type is "live". However, SRT supports the modes that
+the original UDT library supported, that is, *file* and *message* transmission.
 
-There are two general modes: Live and File transmission. Inside File transmission mode, there are also two possibilities: Buffer API
-and Message API. The Live mode uses Message API. However it doesn't
-exactly match the description of the Message API because it uses a maximum
-single sending buffer up to the size that fits in one UDP packet.
+There are two general modes: **Live** and **File** transmission. Inside File 
+transmission mode, there are also two possibilities: **Buffer API** and 
+**Message API**. The Live mode uses Message API. However it doesn't exactly match 
+the description of the Message API because it uses a maximum single sending buffer 
+up to the size that fits in one UDP packet.
 
 There are two options to set a particular type:
 
 * `SRTO_TRANSTYPE`: uses the enum value with `SRTT_LIVE` for live mode
-   and `SRTT_FILE` for file mode. This option actually changes several parameters to 
-   their default values for that mode. After this is done, additional parameters, 
+   and `SRTT_FILE` for file mode. This option actually changes several parameters 
+   to their default values for that mode. After this is done, additional parameters, 
    including those that are set here, can be further changed.
 * `SRTO_MESSAGEAPI`: This sets the Message API (true) or Buffer API (false)
 
 This makes possible a total of three data transmission methods:
 
-* Live
-* Buffer
-* Message
-
-**NOTE THE TERMS** used below:
-* HANGUP and RESUME: "Function HANGS UP" means that it returns
-an error from the `MJ_AGAIN` category (see `SRT_EASYNC*`, `SRT_ETIMEOUT` and
-`SRT_ECONGEST` symbols from `SRT_ERRNO` enumeration type), if it's in
-non-blocking mode. In blocking mode it will block until the condition that
-caused the HANGUP no longer applies, which is defined as that the function
-RESUMES. In nonblocking mode, the function RESUMES when the call to it has done
-something and returned the non-error status. The blocking mode in SRT is
-separate for sending and receiving and set by `SRTO_SNDSYN` and `SRTO_RCVSYN`
-options respectively
-* BLIND REXMIT: A situation where packets that were sent are still not
-acknowledged, either in expected time frame, or when another ACK has
-come for the same number, but no packets have been reported as lost,
-or at least not for all still unacknowledged packets. The congestion control
-class is responsible for the algorithm for taking care of this
-situation, which is either FASTREXMIT or LATEREXMIT. This will be
-expained below.
+* [Live](#transmission-method-live)
+* [Buffer](#transmission-method-buffer)
+* [Message](#transmission-method-message)
+
+### Terminology
+
+The following terms are used in the description of transmission types:
+
+**HANGUP / RESUME**: When a function "HANGS UP" this means that it returns an 
```

Comment:
Actually the intention of this description was to have some common abstract terms for what happens differently depending on whether the function works in blocking and non-blocking mode.

If the function makes a "HANGUP", it means that:
 - in blocking mode, the call will block until the condition no longer applies
 - in non-blocking mode, the call will end up with appropriate "again" error (as enumerated below)

If the function makes a  "RESUME", it means that it has done something it was intended to do, and in blocking mode it exits at all. In case of non-blocking mode it would exit immediately anyway, but in this case with success.

I think better would be to describe these terms both at one for particular mode (separately for blocking and non-blocking).

Reply:
How about this:

**HANGUP**: When a function "HANGS UP" this means that:

  - In *blocking* mode the call will block until the condition no longer applies.
  - In *non-blocking* mode the call will end up with appropriate error. It returns 
  an error from the `MJ_AGAIN`  category (see `SRT_EASYNC*`, `SRT_ETIMEOUT` and 
  `SRT_ECONGEST` symbols from the `SRT_ERRNO` enumeration type.

**RESUME**: When a function "RESUMES" this means that:

  - In *blocking* the function has done something it was intended to do and exits.
  - In *non-blocking* the function has done something and returned a non-error 
  status. It would exit immediately anyway, but in this case successfully.

Note that blocking mode in SRT is separate for sending and receiving, and is set 
by the `SRTO_SNDSYN` and `SRTO_RCVSYN` options respectively.

Reply:
Not sure how clear the "condition" statement is. I'd rather speak about "readiness".

HANGUP is when the function is not ready to perform the operation. In blocking mode it will block, in non-blocking mode will return an error as per this list.

RESUME is when the function is (or better: becomes) ready to perform the operation and it performs it. In both modes the same way.

Reply:
### Terminology

The following terms are used in the description of transmission types:

**HANGUP**: When a function "HANGS UP" this means that it is not ready to perform 
an operation:

  - In *blocking* mode the call will block until the condition no longer applies.
  - In *non-blocking* mode the call will return the appropriate error from the 
  `MJ_AGAIN`  category (see `SRT_EASYNC*`, `SRT_ETIMEOUT` and   `SRT_ECONGEST` 
  symbols from the `SRT_ERRNO` enumeration type.

**RESUME**: When a function "RESUMES" this means that it has become ready to perform 
an operation:

  - In *blocking* mode the function has done something it was intended to do and exits.
  - In *non-blocking* mode the function has done something and returned a non-error 
  status. It would exit immediately anyway, but in this case successfully.

In the descriptions further below, these terms denote the readiness of a function 
under certain conditions. Note that blocking mode in SRT is separate for sending 
and receiving, and is set by the `SRTO_SNDSYN` and `SRTO_RCVSYN` options respectively.